src/datasets.c

```diff
@@ -593,11 +609,45 @@ void DatasetPostReloadCleanup(void)
     SCMutexUnlock(&sets_lock);
 }
 
+static void GetDefaultMemcap(uint64_t *memcap, uint32_t *hashsize) // TODO Do something about this name! OR Save default memcap in dataset struct?
```

Comment:
@victorjulien could you please check if this is a decent way of retrieving defaults from yaml? This shall be called per rule and once during Init.

Reply:
might be able to simplify this to something like:

```c
const char *str = NULL;
if (ConfGetValue("datasets.defaults.memcap", &str) == 1) {
    if (ParseSizeStringU64(str, memcap) < 0) {
        FatalError(SC_ERR_FATAL, "memcap value cannot be deduced: %s", str);
    }
}
```
I think the yaml part is a map, not a list, so we can avoid the loop.

Reply:
OK. Thank you!